pkg/client/block/client.go

```diff
@@ -0,0 +1,169 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"cosmossdk.io/depinject"
+
+	"pocket/pkg/client"
+	"pocket/pkg/either"
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/channel"
+)
+
+var (
+	_ client.BlockClient = (*blockClient)(nil)
+	_ client.Block       = (*cometBlockEvent)(nil)
+)
+
+// blockClient implements the BlockClient interface.
+type blockClient struct {
+	// endpointURL is the URL of RPC endpoint which eventsClient subscription
+	// requests will be sent.
+	endpointURL string
+	// eventsClient is the events query client which is used to subscribe to
+	// newly committed block events. It emits an either value which may contain
+	// an error, at most, once and closes immediately after if it does.
+	eventsClient client.EventsQueryClient
+	// latestBlockObsvblsReplay is a replay observable with replay buffer size 1,
```

Comment:
Would we be able to make it more than 1 or is there a reason to keep it at that value?

Reply:
Probably not for this observable but perhaps the one it emits if we were to change `#LatestBlock()` to `#LastNBlocks()` https://github.com/pokt-network/poktroll/pull/65#discussion_r1372469509. I think this would also require changing the behavior of `ReplayObservable#Last()` to not block until it has `n` replay notifications. I think that's probably a step in the right direction though for `ReplayObservable` generally, I think it makes more sense in most use cases that I can imagine. There are still some details to resolve in that case but I will continue to think about it. :thinking:

Reply:
> Probably not for this observable as we're just using it to hold a reactive reference (only ever a single value) to the observable which emits block events

#PUC

> I think this would also require changing the behavior of ReplayObservable#Last() to not block until it has n replay notification

Tbh, I can see this being a boolean param to `Last(n)` to return if `<n` are available or block if we need `=n`.

Reply:
> I think this would also require changing the behavior of ReplayObservable#Last() to not block until it has n replay notifications. That feels like probably a step in the right direction though for ReplayObservable generally; I think it makes more sense in most use cases that I can imagine. There are still some details to resolve in that case but I will continue to think about it. 🤔

Done.